doc/lua_api.md

```diff
@@ -6732,10 +6732,16 @@ Environment access
     * Generate all registered ores within the VoxelManip `vm` and in the area
       from `pos1` to `pos2`.
     * `pos1` and `pos2` are optional and default to mapchunk minp and maxp.
-* `core.generate_decorations(vm[, pos1, pos2])`
+* `core.generate_decorations(vm[, pos1, pos2, [use_mapgen_biomes]])`
     * Generate all registered decorations within the VoxelManip `vm` and in the
       area from `pos1` to `pos2`.
     * `pos1` and `pos2` are optional and default to mapchunk minp and maxp.
+    * `use_mapgen_biomes` means that decoration generation should
+       respect the biome map of the most recently generated MapChunk,
+       as returned by `get_mapgen_object`.  It is an error to set this
```

Comment:
while it may work the "most recently generated chunk" should not be a supported usage of this function.
it should be valid to use it with the current generation position inside an `on_generated` callback, but never outside of it.

Reply:
we could also encourage callers to do `core.generate_decorations(vm, nil, nil, true)`. this avoids any confusion about positions.

> But in particularly i found it unclear what "it is an error" (like "don't do it"? "it will fail"? )

both, I would assume.